keboola/component_configuration_resource_test.go

```diff
@@ -0,0 +1,256 @@
+		if err != nil {
+			return err
+		}
+		err = checkAttribute("description", storedConfig.Description, attributes["description"])
+		if err != nil {
+			return err
+		}
+		err = checkAttribute("is_disabled", fmt.Sprintf("%v", storedConfig.IsDisabled), attributes["is_disabled"])
+		if err != nil {
+			return err
+		}
+		err = checkAttribute("is_deleted", fmt.Sprintf("%v", storedConfig.IsDeleted), attributes["is_deleted"])
+		if err != nil {
+			return err
+		}
+		err = checkAttribute("name", storedConfig.Name, attributes["name"])
+		if err != nil {
+			return err
+		}
+
+		actualContent := storedConfig.Content
+		expectedContent := orderedmap.New()
+		expectedContentStr := attributes["configuration"]
+		err = expectedContent.UnmarshalJSON([]byte(expectedContentStr))
+		if err != nil {
+			return fmt.Errorf("Could not unmarshal expected configuration to ordered map. Error: %s", err)
+		}
+		actualBytes, err := actualContent.MarshalJSON()
+		if err != nil {
+			return fmt.Errorf("Could not marshal actual configuration to string. Error: %s", err)
+		}
+
+		if !reflect.DeepEqual(actualContent, expectedContent) {
+			return checkAttribute("configuration", string(actualBytes), expectedContentStr)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckExampleConfigurationDataSet(resourceName string, path string, expectedValue string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+		configStr := rs.Primary.Attributes["configuration"]
+		configMap := orderedmap.New()
+		err := configMap.UnmarshalJSON([]byte(configStr))
+		if err != nil {
+			return fmt.Errorf("Couldn't parse configuration: %s", err)
+		}
+		path := orderedmap.PathFromStr(path)
+		value, found, err := configMap.GetNestedPath(path)
+		if err != nil {
+			return fmt.Errorf("Get path failed: %s", err)
+		}
+		if !found {
+			return fmt.Errorf("Get path %s not found:", path)
+		}
+		if value != expectedValue {
+			return fmt.Errorf("Get path %s value didn't match: expected: %s found: %s", path, expectedValue, value)
+		}
+		return nil
+	}
+}
+
+func TestAccConfigResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// create empty config
+			{
+				Config: providerConfig + exGenericResource("testempty", map[string]any{
+					"name": "test empty config",
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkAllAttributesSet("testempty"),
+					resource.TestCheckResourceAttr("keboola_component_configuration.testempty", "name", "test empty config"),
+					resource.TestCheckResourceAttr("keboola_component_configuration.testempty", "is_disabled", "false"),
+					testAccCheckExampleConfigMatchesReality("keboola_component_configuration.testempty"),
+				),
```

Comment:
to je priklad akceptacnych testov. Config je definovany v `Config` property ako concat strings:
```
providerConfig + exGenericResource("testempty", map[string]any{
					"name": "test empty config",
```
Nasledne sa volaju funkcie definovane pod Check property a kontroluje sa terraform state (checkAllAttributesSet, TestCheckResourceAttr) a to co je skutocne ulozene v kbc api (testAccCheckExampleConfigMatchesReality).